script_runner/tasks.py

```diff
@@ -331,27 +340,60 @@ def _prepare_ssl_cert(ssl_cert_content):
 
 class OutputConsumer(object):
 
-    def __init__(self, out):
+    def __init__(self, ctx, out, name):
         self.out = out
-        self.buffer = StringIO()
+        self.name = name
+        self.ctx = ctx
+
+        logs_dir = self._get_log_dir()
+        try:
+            os.makedirs(logs_dir)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+        self.out_file = os.path.join(logs_dir, "{0}.{1}.{2}".format(
+            self.ctx.task_id, self.ctx.operation.retry_number, self.name))
+
         self.consumer = threading.Thread(target=self.consume_output)
         self.consumer.daemon = True
         self.consumer.start()
 
     def consume_output(self):
-        for line in iter(self.out.readline, b''):
-            self.buffer.write(line)
+        with open(self.out_file, 'w') as output:
```

Comment:
i'm not 100% sure, need to consider if this can be `'w'` or if it needs to be `'wb'`
(same for the `'r'` in `ProcessException._get_consumer_file`)

Reply:
@tehasdf you're right, should be `wb`. We can't assume that `stdout` or `stderr` are text. Will fix.